core/types/transaction_marshalling.go

```diff
@@ -94,6 +104,35 @@ func (t *Transaction) MarshalJSON() ([]byte, error) {
 		enc.V = (*hexutil.Big)(tx.V)
 		enc.R = (*hexutil.Big)(tx.R)
 		enc.S = (*hexutil.Big)(tx.S)
+	case *ShutterTx:
+		enc.ChainID = (*hexutil.Big)(tx.ChainID)
+		enc.Nonce = (*hexutil.Uint64)(&tx.Nonce)
+		enc.Gas = (*hexutil.Uint64)(&tx.Gas)
+		enc.MaxFeePerGas = (*hexutil.Big)(tx.GasFeeCap)
+		enc.MaxPriorityFeePerGas = (*hexutil.Big)(tx.GasTipCap)
+		enc.EncryptedPayload = (*hexutil.Bytes)(&tx.EncryptedPayload)
+		enc.L1BlockNumber = (*hexutil.Uint64)(&tx.L1BlockNumber)
+		enc.BatchIndex = (*hexutil.Uint64)(&tx.BatchIndex)
+		enc.To = t.To()
```

Comment:
`ShutterTxs` don't have a (plaintext) `To` field.

Reply:
see above